Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py

```diff
@@ -461,6 +468,49 @@ def add_comment(self, issue, text):
 
         return result
 
+    def relate(self, issue, depends_on=None, blocks=None, regressed_by=None, regressions=None, **relations):
+        if relations:
+            raise TypeError("'{}' is an invalid relation".format(list(relations.keys())[0]))
+
+        update_dict = dict()
+        update_dict['ids'] = [issue.id]
+        if depends_on:
```

Comment:
This should probably be an `elif`, right? Also, more generally, it's not correct to try and define multiple relationships at a time, is it? If it is, then this function should really accept lists of issues, if it's not, then we should have code that validates the user only passed a single issue. That probably points to accepting `**relations` and then asserting there is only a single key, then validating the key is one of the expected inputs.

Reply:
The API supports setting multiple relationships in one request, so I wanted to add that as an option. issue should be a singular issue, whereas the keyword args are all supposed to be lists of issues. Should I add type hints to make that clear?

Reply:
We still have Python 2 support here, so we have to be careful with type hints.

Because the API supports setting multiple, this is reasonable, although it is probably better to specifically iterate over the arguments and extract IDs rather than expecting the caller to pass integer IDs instead of `Issue` objects. Might also want to check that the `Issue` objects in question are actually from this tracker as well....